### ISSUE TITLE
[Bug]: `min_tokens` and `ignore_eos` sampling parameters not enforced in TPU JAX sampler

### ISSUE THREAD

**Issue by shireen-bean** [vllm-project/tpu-inference#1524]
### Your current environment

- **Docker Image**: `vllm/vllm-tpu:nightly`
- **Image SHA**: `sha256:b4073cd1409275255e66d32492b74b54bd1f9b994c5f90f28d49a0d93dd62e04`
- **vLLM Version**: `0.14.0rc2.dev265+g7ef587375`
- **vLLM Commit**: [`7ef5873`](https://github.com/vllm-project/vllm/commit/7ef587375280b475719c37463ca37f2ceb38e54e)
- **tpu-inference Commit**: [`ba1595a`](https://github.com/vllm-project/tpu-inference/commit/ba1595a49eb6aca94065e5fb2f2615d29007caac)
- **TPU Type**: v6e-1 (1x1)

### Description

The `min_tokens` and `ignore_eos` sampling parameters are accepted by the OpenAI-compatible API but are not enforced during token generation on TPU. This causes requests to terminate early when the model generates an EOS token, regardless of the `min_tokens` setting.

### Environment
- vLLM TPU image: `vllm/vllm-tpu:nightly`
- TPU type: v6e-1
- vLLM version: (from container)

### Steps to Reproduce

1. Start vLLM on TPU with a model
2. Send a chat completion request with `min_tokens=30` and `ignore_eos=true`:

```
curl http://localhost:8000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{
    "model": "your-model",
    "messages": [{"role": "user", "content": "Hello"}],
    "max_tokens": 30,
    "min_tokens": 30,
    "ignore_eos": true
  }'
 ```
 
3. Response often contains fewer than 30 tokens when the model generates an EOS-like token early### Expected Behavior

Generation should continue until at least `min_tokens` tokens are produced, with EOS tokens masked/ignored until that threshold is reached.

### Actual Behavior

Generation stops immediately when an EOS token is sampled, typically producing only 5-6 tokens instead of the requested 30.

### Root Cause Analysis

After investigating the source code, I found:

**GPU implementation (working):**
- `vllm/v1/sample/logits_processor/builtin.py` contains `MinTokensLogitsProcessor`
- This processor sets stop token logits to `-inf` until `min_tokens` is reached
- Applied via `apply_logits_processors()` in the GPU sampler

**TPU implementation (missing):**
- `tpu_inference/layers/jax/sample/sampling.py` - the `sample()` function only applies:
  - Top-k masking
  - Top-p masking
  - Temperature scaling
- **No `min_tokens` enforcement or EOS masking**

The `min_tokens` value IS stored in `tpu_inference/runner/input_batch.py`:
# req_index -> (min_tokens, stop_token_ids)
self.min_tokens: dict[int, tuple[int, set[int]]] = {}But this dict is **never used** in the JAX sampling path.

The `check_stop()` function in `vllm/v1/core/sched/utils.py` does check `min_tokens`, but this runs **after** sampling - by then the EOS token has already been generated.

### Additional Context

This issue is exacerbated when `--enable-prefix-caching` is used:
- First ~50 requests work correctly (before cache is populated)
- Subsequent requests hit cached KV states that include the EOS decision
- Results in 97%+ of requests generating only 5 tokens despite `min_tokens=30`

Disabling prefix caching is a partial workaround but doesn't fix the underlying issue.

### Evidence in the wild

The figure shows latency CDFs comparing TPU (orange) vs GPU (blue) at 10, 20, 50, and 75 QPS.
Token counts are annotated at each percentile.

**Observation**: At low load (10-20 QPS), both platforms respect `min_tokens=30`. 
At higher load (50-75 QPS), TPU degrades to generating only 5 tokens (P50-P90) 
while GPU maintains ~30 tokens. This suggests the bug is exacerbated by 
prefix caching or high-concurrency scheduling

<img width="2083" height="1475" alt="Image" src="https://github.com/user-attachments/assets/88ea0717-21b8-47cc-9605-f47b3d7dd624" />

vLLM on TPU args used:

```
      command: ["python3", "-m", "vllm.entrypoints.openai.api_server"]
      args:
            - '--gpu-memory-utilization=0.95'
            # Disabled: prefix caching interferes with ignore_eos/min_tokens
            # - '--enable-prefix-caching'
            - '--max-model-len=8192'
            - '--max-logprobs=60'
            - '--max-num-seqs=64'
            - '--max-num-batched-tokens=8192'
            - '--enable-chunked-prefill'
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues and checked the [documentation page](https://github.com/vllm-project/tpu-inference/tree/main/docs), which can answer lots of frequently asked questions.

**Comment by JiriesKaileh** [comment]
Please try the fix implemented [here](https://github.com/vllm-project/tpu-inference/commit/b6cf7923bf596baee9eb0e0c55a849b673a0c833). I was having trouble reproducing the error on my end, so let me know if the attempted fix does not resolve the issue.

**Comment by shireen-bean** [comment]
Thanks @JiriesKaileh ! I'm all set here. TPU JAX sampler still lacks a MinTokensLogitsProcessor equivalent that masks EOS logits but nolonger blocked! thanks!
